src/Run.hs

```diff
@@ -57,10 +59,10 @@ runWeb args = do
     reportSocketName
     Http.start socketName $ withMVar lock $ \() -> trigger interpreter
 
-trigger :: Session -> IO String
+trigger :: Session -> IO (Bool, String)
 trigger interpreter = do
   xs <- Interpreter.reload interpreter
-  ys <- if "Ok, modules loaded:" `isInfixOf` xs
+  (ys, summary) <- if "Ok, modules loaded:" `isInfixOf` xs
     then Interpreter.hspec interpreter
-    else return ""
-  return (xs ++ ys)
+    else return ("", Nothing)
+  return (maybe False ((== 0) . summaryFailures) summary, xs ++ ys)
```

Comment:
So the summary may be `Nothing` when parsing the `Summary` fails? And we interpret that as failure?